### PR TITLE
fix: security is a dictonary

### PR DESCRIPTION
**Pull request by XVincentX**
This is probably another breaking change :(

### CODE REVIEW

File: src/http-spec.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface IHttpService extends INode {
   };
   version: string;
   servers: IServer[];
-  security: string[];
+  security: Dictionary<string[], string>;
```
Comment (lottamus) [inline src/http-spec.ts]: Should we just use the same typing as HttpOperation security? https://github.com/stoplightio/types/pull/33/files#diff-6b2dc2c69dd576eb785765cc8d3bb904L38

Comment (XVincentX) [reply src/http-spec.ts]: In theory yes, they are the same. However the problem I just spotted is that `HttpSecurityScheme` is *wrong* for `security`

```ts

export interface IApiKeySecurityScheme extends ISecurityScheme {
  type: 'apiKey';
  name: string;
  in: 'query' | 'header' | 'cookie';
}

export interface IBearerSecurityScheme extends ISecurityScheme {
  type: 'http';
  scheme: 'bearer';
  bearerFormat?: string;
}

export interface IBasicSecurityScheme extends ISecurityScheme {
  type: 'http';
  scheme: 'basic';
}

export interface IOpenIdConnectSecurityScheme extends ISecurityScheme {
  type: 'openIdConnect';
  openIdConnectUrl: string;
}

export interface IOauth2SecurityScheme extends ISecurityScheme {
  type: 'oauth2';
  flows: IOauthFlowObjects;
}
```

This is not what's security is about. This is what `securityDefinitions` is about :(

I'm digging into and trying to fix this.

Comment (XVincentX) [reply src/http-spec.ts]: @lottamus Does that make sense?

Comment (lottamus) [reply src/http-spec.ts]: Right but I think the idea was to denormalize the http operation has the full security scheme object. I think we could just do the same for http service too

See how it's being translated here: https://github.com/stoplightio/graphite/blob/e5c93d777fde1f434cf9d318e4e9cc8711da5301/src/plugins/http/oas2/translators/operation.ts#L37

Comment (XVincentX) [reply src/http-spec.ts]: I see…this is **so** confusing though.

Comment (lottamus) [reply src/http-spec.ts]: What's confusing? Http Operation and Http Service don't need to represent the OpenAPI specs _exactly_.  However I think they should denormalize data between all versions of OpenAPI in a consistent way.

Comment (XVincentX) [reply src/http-spec.ts]: > What's confusing? 

The main source of confusion to be is that the service has a shape that reminds a lot the OpenAPI 2 spec and to you're inclined to mimic its content.

I take a lot of blame on that; basically I missed the opportunity of giving better names — for example `globalSecurityDefinitions` would have been **way** better than `security` that recalls OAS2 by default.

In any case I think I _finally_ get what's going on. Is the PR good to go or did I do another disaster?

**Pull request merged**